### PR TITLE
Implement DELETE /v3/apps/:guid

### PR DESCRIPTION
**Pull request by davewalter**
## Is there a related GitHub Issue?
#334

## What is this change about?
This PR implements the `DELETE /v3/apps/:guid` endpoint for the `cf delete` command.

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See #334

## Tag your pair, your PM, and/or team
@Birdrock

### CODE REVIEW

File: api/apis/app_handler.go [hunk 1]
```diff
@@ -659,6 +661,48 @@ func (h *AppHandler) appRestartHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *AppHandler) appDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	appGUID := vars["guid"]
+
+	authInfo, ok := authorization.InfoFromContext(ctx)
```
Comment (tcdowney) [inline api/apis/app_handler.go:669]: Nice, glad to see we're using this in new endpoints 🙂

File: api/repositories/app_repository.go [hunk 2]
```diff
@@ -363,6 +368,17 @@ func (f *AppRepo) SetAppDesiredState(ctx context.Context, authInfo authorization
...
 
+func (f *AppRepo) DeleteApp(ctx context.Context, authInfo authorization.Info, message AppDeleteMessage) error {
+	cfApp := &workloadsv1alpha1.CFApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      message.AppGUID,
+			Namespace: message.SpaceGUID,
+		},
+	}
+
+	return f.privilegedClient.Delete(ctx, cfApp)
```
Comment (tcdowney) [inline api/repositories/app_repository.go:379]: Would it work still if we used the user client like we do in AppList?

https://github.com/cloudfoundry/cf-k8s-controllers/pull/360/files#diff-433641cf784e6cc94960970a9af8ae3a9c392337e21bc68d90485dd5c9a31b10R200

**Review by tcdowney — APPROVED** [review]
Overall looks good to me. There does look like an opportunity here to use a user client instead of the privileged client to perform the delete, but I'd understand if you'd rather tackle that more holistically.

Eventually I think we want to implement authz as part of building out a new endpoint, but we haven't really discussed that much. I was just inspired by the AppRepo constructor. 😛

**Review by akrishna90 — COMMENTED** [review]
@Birdrock - update the api docs!

File: api/repositories/app_repository.go (hunk 2 above)
Comment (davewalter) [reply api/repositories/app_repository.go]: @tcdowney It works with some minor RBAC tweaks.

**Pull request merged**